### PR TITLE
CK Editor Beta Launch

### PR DESCRIPTION
**Pull request by Raemon**
cleans up some issues with a second CkEditor for posts, which enables users to choose whether to enable collaboration with other users.

(fixes a bug where previously the editor wouldn't re-render with the collaborative settings if you switched to it from the first version of CKEditor)

### CODE REVIEW

File: public/lesswrong-editor/src/ckeditor.js [hunk 1]
```diff
@@ -125,21 +125,18 @@ const postEditorConfig = {
...
 	image: {
 		toolbar: [
-      'imageStyle:alignLeft',
-      'imageStyle:alignCenter',
-      'imageStyle:alignRight',
-      'imageStyle:full',
-			'|',
-			'imageTextAlternative',
-			'|',
+      // 'imageStyle:alignLeft',
```
Comment (jimrandomh) [inline public/lesswrong-editor/src/ckeditor.js]: Why are these commented out now?

File: packages/lesswrong/themes/stylePiping.js [hunk 2]
```diff
@@ -238,11 +241,14 @@ export const ckEditorStyles = theme => {
         },
         '& .ck-annotation__info-name, & .ck-annotation__info-time, & .ck-comment__input, & .ck-thread__comment-count, & .ck-annotation__main p, & .ck-annotation__info-name, & .ck-annotation__info-time, & .ck-presence-list__counter, &.ck-presence-list': {
           ...commentBodyStyles(theme),
+          marginTop: -10,
```
Comment (jimrandomh) [inline packages/lesswrong/themes/stylePiping.js]: Negative margins are generally concerning. Is there another margin that this is cancelling out, which could be removed instead?

File: packages/lesswrong/lib/collections/posts/custom_fields.js [hunk 3]
```diff
@@ -72,8 +72,9 @@ addFieldsDict(Posts, {
   title: {
     order: 10,
     placeholder: "Title",
+    label: "Title",
     control: 'EditTitle',
-    editableBy: [Users.owns, 'sunshineRegiment', 'admins']
+    editableBy: ['sunshineRegiment', 'admins']
```
Comment (jimrandomh) [inline packages/lesswrong/lib/collections/posts/custom_fields.js]: Eek, no, this means users can no longer edit the titles of their posts.

File: packages/lesswrong/lib/collections/posts/collection.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ const options = {
         Users.canDo(user, 'posts.alignment.suggest')) {
       return true
     }
-    return Users.owns(user, document) ? Users.canDo(user, 'posts.edit.own') : Users.canDo(user, `posts.edit.all`)
+    return Posts.canEdit(user, document) //Users.owns(user, document) ? Users.canDo(user, 'posts.edit.own') : Users.canDo(user, `posts.edit.all`)
```
Comment (jimrandomh) [inline packages/lesswrong/lib/collections/posts/collection.js]: Remove debris

File: packages/lesswrong/components/posts/PostsPage/PostActions.jsx [hunk 5]
```diff
@@ -140,6 +140,16 @@ class PostActions extends Component {
             Edit
           </MenuItem>
         </Link>}
+        { Users.canCollaborate(currentUser, post) && 
+          <Link to={{pathname:'/collaborateOnPost/', search:`?${qs.stringify({postId: post._id})}`}}>
```
Comment (jimrandomh) [inline packages/lesswrong/components/posts/PostsPage/PostActions.jsx]: URL shouldn't have a trailing slash.
```suggestion
          <Link to={{pathname:'/collaborateOnPost', search:`?${qs.stringify({postId: post._id})}`}}>
```

File: packages/lesswrong/lib/collections/posts/custom_fields.js (hunk 3 above)
Comment (Raemon) [reply packages/lesswrong/lib/collections/posts/custom_fields.js]: Sorry, that was debris while trying to generally experiment with "can I make the editableBy function do anything on purpose". Reset to it's original value now

File: public/lesswrong-editor/src/ckeditor.js (hunk 1 above)
Comment (Raemon) [reply public/lesswrong-editor/src/ckeditor.js]: Oli didn't want image positioning to avoid users creating weirdly-formatted-mobile content

Comment (Raemon) [reply public/lesswrong-editor/src/ckeditor.js]: (deleted the comments now)

**Comment by Discordius** [comment]
Waiting on lint fix, then it seems good to go

**Pull request merged**